### PR TITLE
Added stylelint and stylefmt support

### PR DESCRIPTION
**Pull request by amilajack**
- [x] Added stylelint and stylefmt support
- [x] Conformed all styles to convention

### CODE REVIEW

File: .stylelintrc [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "extends": "stylelint-config-standard",
+	"rules": {
+		"color-hex-case": "upper",
```
Comment (mcnuttandrew) [inline .stylelintrc]: i kinda prefer lower cased hexes personally

File: .stylelintrc [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+    "selector-list-comma-newline-after": "never-multi-line"
```
Comment (mcnuttandrew) [inline .stylelintrc]: also i prefer multi-lining multi-selectors

File: package.json [hunk 3]
```diff
@@ -23,6 +23,8 @@
     "clean": "rm -rf dist bundle.* index.html && mkdir dist",
     "build": "npm run clean && babel src -d dist --copy-files && node-sass src/main.scss main.css",
     "lint": "eslint src tests showcase --ignore-pattern node_modules --ignore-pattern bundle.js",
+    "lint-styles": "stylelint examples/*.scss showcase/*.scss src/*.scss src/styles/*.scss --syntax scss",
+    "lint-styles-fix": "stylefmt -r examples/*.scss showcase/*.scss src/*.scss src/styles/*.scss --syntax scss",
```
Comment (mcnuttandrew) [inline package.json]: i'm not sure i understand what this does?

File: examples/examples.scss [hunk 4]
```diff
@@ -7,7 +7,7 @@ body {
   font-size: 14px;
 }
 
-h1, h2, h3, h4, h5 {
+h1,h2,h3,h4,h5 {
```
Comment (Apercu) [inline examples/examples.scss]: What's the purpose of this one? Minifying our source?

File: .stylelintrc (hunk 1 above)
Comment (amilajack) [inline .stylelintrc]: The convention in the project was mixed so i wasnt sure what to choose. I'll fix this

File: .stylelintrc [hunk 5]
```diff
@@ -0,0 +1,7 @@
+{
+  "extends": "stylelint-config-standard",
+	"rules": {
```
Comment (Apercu) [inline .stylelintrc]: Can you format this config instead of keeping mixed tab and spaces?

File: package.json (hunk 3 above)
Comment (Apercu) [inline package.json]: Kinda the same as `eslint --fix`

Comment (Apercu) [inline package.json]: Lot of repetition here, can't we avoid that? Also unsure if we need to add `stylefmt`, it's useful the first time, but not so much after.

File: package.json [hunk 6]
```diff
@@ -61,6 +63,9 @@
     "react-addons-shallow-compare": ">=15.4.2",
     "react-addons-test-utils": ">=15.4.2",
     "react-dom": ">=15.4.2",
+    "stylefmt": "^5.1.1",
+    "stylelint": "^7.7.1",
+    "stylelint-config-standard": "^15.0.1",
```
Comment (Apercu) [inline package.json]: Note: there is no such thing as standard in the js world 😁

File: .stylelintrc (hunk 1 above)
Comment (Apercu) [inline .stylelintrc]: Same

File: examples/examples.scss [hunk 7]
```diff
@@ -57,16 +57,19 @@ nav {
   float: right;
   padding: 0;
   font-weight: 400;
+
```
Comment (Apercu) [inline examples/examples.scss]: Do we want this @mcnuttandrew ?

File: src/styles/plot.scss [hunk 8]
```diff
@@ -1,36 +1,28 @@
...
-  color: #c3c3c3;
+  color: #C3C3C3;
   position: relative;
 }
 
 .rv-xy-plot__inner {
   display: block;
 }
 
-.rv-xy-plot__axis {
```
Comment (Apercu) [inline src/styles/plot.scss]: I have a feeling this was here as a reminder

File: showcase/index.scss [hunk 9]
```diff
@@ -1,4 +1,4 @@
-@import "../main";
+@import "../src/main.scss";
```
Comment (Apercu) [inline showcase/index.scss]: Should keep this as it is

Comment (Apercu) [inline showcase/index.scss]: Hum actually not sure now, maybe it's fine

**Comment by Apercu** [comment]
The showcase stylesheet and example one are exactly the same, we should probably want to merge them.

**Comment by amilajack** [comment]
What about creating a `examples.scss` file in the root and removing the existing `./showcase/index.scss` and `./examples/examples.scss`?

**Comment by Apercu** [comment]
Not a fan of throwing too much files in the root, it should be clean. `src/styles` would be a better place.

**Comment by amilajack** [comment]
Removed duplication by creating a single `./src/styles/examples.scss`

**Comment by Apercu** [comment]
Nice! ;) And could you remove the two rule overrides?

**Comment by amilajack** [comment]
Which rules are you referring to?

**Comment by amilajack** [comment]
I added the second rule because of @mcnuttandrew 's [comment](https://github.com/uber/react-vis/pull/255#discussion_r97651011)

**Comment by Apercu** [comment]
My bad, keep the multi-line, but remove the uppercase

File: src/styles/plot.scss [hunk 10]
```diff
@@ -1,36 +1,30 @@
...
-$rv-xy-plot-axis-line-color: #e6e6e9;
+$rv-xy-plot-axis-line-color: #E6E6E9;
 $rv-xy-plot-axis-font-size: 11px;
-
 $rv-xy-plot-tooltip-background: #3A3A48;
-$rv-xy-plot-tooltip-color: #fff;
+$rv-xy-plot-tooltip-color: #FFF;
 $rv-xy-plot-tooltip-font-size: 12px;
 $rv-xy-plot-tooltip-border-radius: 4px;
 $rv-xy-plot-tooltip-shadow: 0 2px 4px rgba(0, 0, 0, 0.5);
```
Comment (mcnuttandrew) [inline src/styles/plot.scss]: while were linting stuff, prefer rbga with hexes: rgba(#000, 0.5) would be better here

Comment (amilajack) [inline src/styles/plot.scss]: Should this be done for all hexes or just this one?

File: package.json [hunk 11]
```diff
@@ -23,6 +23,7 @@
     "clean": "rm -rf dist bundle.* index.html && mkdir dist",
     "build": "npm run clean && babel src -d dist --copy-files && node-sass src/main.scss main.css",
     "lint": "eslint src tests showcase --ignore-pattern node_modules --ignore-pattern bundle.js",
+    "lint-styles": "stylelint src/*.scss src/styles/*.scss --syntax scss",
```
Comment (Apercu) [inline package.json]: We should probably be fine with only `stylelint src/styles/*.scss --syntax scss`

Comment (amilajack) [inline package.json]: Fixed

**Comment by Apercu** [comment]
Nice 😃

**Pull request merged**